sendgrid/sendgrid.py

```diff
@@ -58,3 +57,14 @@ def api_key(self, value):
     @property
     def impersonate_subuser(self):
         return self._impersonate_subuser
+
+
+class SendGridClient(SendGridAPIClient):

```

Comment:
I believe this is why @w- is suggesting we handle the exceptions in `python_http_client.Client`.